File: 0229-majority-element-ii/0229-majority-element-ii.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        hashmap = {}

        for num in nums:
            if hashmap.get(num):
                hashmap[num] += 1
            else:
                hashmap[num] = 1
        
        ans = []
        for h, s in hashmap.items():
            if s > len(nums)/3:
                ans.append(h)

        return ans

File: 0229-majority-element-ii/0229-majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        Map<Integer, Integer> hashmap = new HashMap<>();

        for (int num : nums) {
            if (hashmap.containsKey(num)) {
                hashmap.put(num, hashmap.get(num) + 1);
            } else {
                hashmap.put(num, 1);
            }
        }

        ArrayList<Integer> arr = new ArrayList<>();
        hashmap.forEach((key, value) -> {
            if (value > nums.length / 3) {
                arr.add(key);
            }
        });

        return arr;
    }
}